### ISSUE TITLE
Empty base URL results in strange behavior with non-default ports

### ISSUE THREAD

**Issue by etiago** [Microsoft/typed-rest-client#66]
### Environment
Node version: 8.10
Npm version: 5.8.0
OS and version: MacOS 10.13.3
typed-rest-client version: stable

### Issue Description
Using Typed Rest Client without specifying a base URL does not play well with non-default ports. I was using a node server on port 3000 and without specifying a URL, it tries to send requests as follows:
```
http://localhost:3000/:80/product/abca
```

### Expected behaviour
I expect it to behave like the `fetch` polyfill, where as if a base URL is not specified, it takes the request URL as an absolute URL for the current host:port combination. In my specific case, when I do a request to `/product/abca`, I expect the request to go to `http://localhost:3000/product/abca`.

### Actual behaviour
Currently it seems to be a weird hybrid as since I didn't specify a base URL and my requests are relative, it appends the port and probably has an empty hostname. Basically currently we have a strange behavior, which possibly should throw an error if we don't want to support empty base URL + relative requests.

### Steps to reproduce
1. let restClient: rc.RestClient = new rc.RestClient('frontend');
2.  let response: rc.IRestResponse<Array<Product>> = await restClient.get<Array<Product>>('/product/' + searchString);

### Logs
Not relevant.

**Comment by etiago** [comment]
I really like the idea behind `typed-rest-client` but as it stands it makes it unusable for a backend+frontend combo unless I hardcode the host where my application is running (which can vary per deployment). Ideally I would like an HTTP client which will just take the current host and pick it up from there when using relative paths.

**Comment by stephenmichaelf** [comment]
Thanks for reaching out about this!

I want to make sure I understand: you are saying when you have the library running on a machine you want it to use the url of the machine it's running on as the base if you don't pass one in? So it can call in to itself? Can't you dynamically get the URL at runtime and set it before you make your request? Or if you want you could do a PR with the change :)

**Comment by etiago** [comment]
Now I realize I didn't explain the problem I'm getting as well as I thought I did...!

The issue is specific to running the library as part of an application running in the browser. Specifically I'm seeing it in the context of a React application. I think this is relevant because when running in the browser, if I'm sending a request to `/products`, I expect it to go to `<WHATEVER_DOMAIN_THE_WEBSITE_IS_BEING_ACCESSED_ON_USING_WHATEVER_PROTOCOL_ON_WHATEVER_PORT>/products`. With an example, if I'm visiting `https://www.freshfruit.com/` and if within the JS I make a request to `/products`, I expect the request to go to `https://www.freshfruit.com/products`.

This behavior is likely to be different in a Node backend application as then there's no implicit `baseUrl` such as the one in the browser.

Nonetheless, I've submitted a pull request with a change that fixes my issue. The existing tests also all currently pass. If we're happy with the change, I would also like to write a couple more tests to go with it (and I will need to sort out a CLA first as I'm not sure my employer contractually currently allows me to share any code publicly).
